subprojects/core-api/src/main/java/org/gradle/work/InputChanges.java

```diff
@@ -98,5 +99,18 @@
      *
      * @param parameterValue The value of the parameter to query.
      */
-    Iterable<FileChange> getFileChanges(Object parameterValue);
+    Iterable<FileChange> getFileChanges(FileCollection parameterValue);
+
+    /**
+     * Changes for a parameter.
```

Comment:
I think it'd be important to mention that a `DirectoryProperty` and a `RegularFileProperty` are `Provider<? extends FileSystemLocation>`s. These two types will be the ones typically passed in here, and their relationship to `Provider<? extends FSL>` might not be obvious to the user.

Reply:
Mentioned.